plugin-annotation/src/main/java/com/mapbox/maps/plugin/annotation/AnnotationManagerImpl.kt

```diff
@@ -338,31 +343,42 @@ abstract class AnnotationManagerImpl<G : Geometry, T : Annotation<G>, S : Annota
         Logger.e(TAG, "Can't update source: source has not been added to style.")
         return
       }
-      annotations
-        .filter { it.getType() == AnnotationType.PointAnnotation }
-        .forEach {
-          val symbol = it as PointAnnotation
-          symbol.iconImage?.let { image ->
-            if (image.startsWith(PointAnnotation.ICON_DEFAULT_NAME_PREFIX)) {
-              // User set the bitmap icon, add the icon to style
-              symbol.iconImageBitmap?.let { bitmap ->
-                if (style.getStyleImage(image) == null) {
-                  val imagePlugin = image(image) {
-                    bitmap(bitmap)
+      workerHandler.post {
+        val lock = ReentrantLock()
+        val waitCondition = lock.newCondition()
+        lock.withLock {
+          annotations
+            .filter { it.getType() == AnnotationType.PointAnnotation }
+            .forEach {
+              val symbol = it as PointAnnotation
+              symbol.iconImage?.let { image ->
+                if (image.startsWith(PointAnnotation.ICON_DEFAULT_NAME_PREFIX)) {
+                  // User set the bitmap icon, add the icon to style
+                  symbol.iconImageBitmap?.let { bitmap ->
+                    if (style.getStyleImage(image) == null) {
+                      val imagePlugin = image(image) {
+                        bitmap(bitmap)
+                      }
+                      style.addImage(imagePlugin)
+                    }
                   }
-                  style.addImage(imagePlugin)
                 }
               }
             }
+          val features = annotations.map {
+            val annotation = Feature.fromGeometry(it.geometry, it.getJsonObjectCopy())
+            it.setUsedDataDrivenProperties()
+            annotation
           }
+
+          geoJsonSource.featureCollection(
+            FeatureCollection.fromFeatures(features),
+            onDataParsed = { lock.withLock { waitCondition.signal() } }
+          )
+          // wait until the former update is completed.
+          waitCondition.await()
```

Comment:
Although it should work more or less OK now (apply all the updates) - I bet we can hit another problem: if interacting with map and annotations for pretty long time user might see increasing delay between actually sending annotation data to the map and seeing it on the map.
Could you try to simulate such scenario? IDK perhaps drag some annotation over the map for long period of time. If it's a problem perhaps we might want to add some API to `update` function whether to apply __all__ changes (but with possible delay) or apply latest changes only.

Reply:
If there are too many annotations(>2000), there would be a delay, but I'm not seeing an increasing delay, this is due to the bad performance for annotation plugin for large-scale annotations.

Reply:
ok, thanks for confirming. 
if needed - we could extend this API later.